api/src/tests/transactions_test.rs

```diff
@@ -1504,6 +1504,44 @@ async fn test_simulation_failure_error_message() {
         .contains("Division by zero"));
 }
 
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_simulation_failure_with_detail_error() {
+    let mut context = new_test_context(current_function_name!());
+    let account = context.root_account().await;
+    let raw_txn = context
+        .transaction_factory()
+        .entry_function(EntryFunction::new(
+            ModuleId::new(
+                AccountAddress::from_hex_literal("0x1").unwrap(),
+                Identifier::new("MemeCoin").unwrap(),
+            ),
+            Identifier::new("transfer").unwrap(),
+            vec![APTOS_COIN_TYPE.clone()],
+            vec![
+                bcs::to_bytes(&AccountAddress::from_hex_literal("0xdd").unwrap()).unwrap(),
+                bcs::to_bytes(&1u64).unwrap(),
+            ],
+        ))
+        .sender(account.address())
+        .sequence_number(account.sequence_number())
+        .expiration_timestamp_secs(u64::MAX)
+        .build();
+    let invalid_key = AccountKey::generate(&mut context.rng());
+    let txn = raw_txn
+        .sign(invalid_key.private_key(), account.public_key().clone())
+        .unwrap()
+        .into_inner();
+    let body = bcs::to_bytes(&txn).unwrap();
+    let resp = context
+        .expect_status_code(200)
+        .post_bcs_txn("/transactions/simulate", body)
+        .await;
+    assert!(resp.as_array().unwrap()[0]["vm_status"]
```

Comment:
Will this test fail on main without your patch?

Reply:
it fails. This the resp without the change     "vm_status": "Execution failed with miscellaneous error and no status code\nExecution failed with message: Linker Error: Module 0000000000000000000000000000000000000000000000000000000000000001::MemeCoin doesn't exist\nBacktrace: . And the resp with the change `vm_status":"LINKER_ERROR\nExecution failed with message: Linker Error: Module 0000000000000000000000000000000000000000000000000000000000000001::MemeCoin doesn't exist\nBacktrace: Backtrace `

Reply:
Beautiful!